homeassistant/components/apsystems/sensor.py

```diff
@@ -111,43 +108,36 @@ class ApsystemsLocalApiSensorDescription(SensorEntityDescription):
 
 async def async_setup_entry(
     hass: HomeAssistant,
-    config_entry: ConfigEntry,
+    config_entry: ApSystemsConfigEntry,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the sensor platform."""
     config = config_entry.runtime_data
-    device_id = config_entry.unique_id
-    assert device_id
 
     add_entities(
-        ApSystemsSensorWithDescription(config, desc, device_id) for desc in SENSORS
+        ApSystemsSensorWithDescription(
+            data=config,
+            entity_description=desc,
+        )
+        for desc in SENSORS
     )
 
 
-class ApSystemsSensorWithDescription(
-    CoordinatorEntity[ApSystemsDataCoordinator], SensorEntity
-):
+class ApSystemsSensorWithDescription(ApSystemsCoordinatorEntity, SensorEntity):
```

Comment:
If we agree to drop `ApSystemsCoordinatorEntity`, then I think this entity should inherit all three
```suggestion
class ApSystemsSensorWithDescription(ApSystemsEntity, CoordinatorEntity[ApSystemsDataCoordinator], SensorEntity):
```